src/HookTrait.php

```diff
@@ -58,29 +58,19 @@ public function onHook($spot, $fx = null, $args = [], $priority = 5)
 
         // short for onHook('test', $this); to call $this->test();
         if (!is_callable($fx)) {
+            $valid = false;
             if (is_object($fx)) {
-                if (isset($fx->_dynamicMethodTrait)) {
-                    if (!$fx->hasMethod($spot)) {
-                        throw new Exception([
-                            '$fx should be a valid callback',
-                            'fx' => $fx,
-                        ]);
-                    }
-                } else {
-                    if (!method_exists($fx, $spot)) {
-                        throw new Exception([
-                            '$fx should be a valid callback',
-                            'fx' => $fx,
-                        ]);
-                    }
-                }
-                $fx = [$fx, $spot];
-            } else {
+                $valid = isset($fx->_dynamicMethodTrait) && $fx->hasMethod($spot) || method_exists($fx, $spot);
```

Comment:
what do you think about addingbrackets to clarify logic?
`$valid = (isset($fx->_dynamicMethodTrait) && $fx->hasMethod($spot)) || method_exists($fx, $spot);`

Reply:
👍 everything that makes code speak for itself